94 add extra endpoints for 52

CristiSlv1 opened the pull request.

CristiSlv1 committed:
Add course management methods to IRepository

Updated `IRepository.cs` to include new async methods for enrolling users, checking enrollment status, verifying course completion, updating time spent, and claiming rewards.

Implemented these methods in `Repository.cs`, enhancing course management capabilities with logic for user enrollment, course completion checks, time tracking, and filtered course retrieval based on various criteria.

CristiSlv1 committed:
Add CourseCompletion management to DataContext

- Introduced `DbSet<CourseCompletion>` for tracking course completions.
- Configured `CourseCompletion` entity with a composite key of `UserId` and `CourseId` to establish a one-to-many relationship.

CristiSlv1 committed:
Add new API endpoints to CourseController

Implemented methods for user enrollment, course filtering,
status checks, time tracking, and reward claims.
Included a new using directive for System.Linq.

CristiSlv1 committed:
Add migration for CourseCompletions tracking

Introduces a new migration class `AddCourseCompletions` to manage the database schema for course completions. Updates the existing `CourseMigration` to reflect the new migration order. Adds the `CourseCompletion` entity to the `DataContextModelSnapshot`, including properties and relationships with `Course` and `User`. Ensures the database schema supports tracking user progress in courses.

UjfalusiAbel commented on Duo.Api/Controllers/CourseController.cs:
```diff
@@ -80,5 +81,135 @@ public async Task<IActionResult> DeleteCourse([FromQuery] int id)
+            return Ok(isCompleted);
+        }
+
+        [HttpGet("get-filtered")]
```
Move this below the comments, on top of the function

UjfalusiAbel reviewed (changes requested):
Please change that one line

CristiSlv1 committed:
Add GetFilteredCourses endpoint to CourseController

This commit introduces a new HTTP GET endpoint `GetFilteredCourses` in the `Duo.Api.Controllers.CourseController` class. The endpoint allows filtering of courses based on search text, course type (premium or free), enrollment status, and selected tags. Additionally, detailed XML documentation comments have been added to improve code readability and usability for developers.
